enclave/sgx/revocationinfo.c

```diff
@@ -3,161 +3,170 @@
 
 #include <openenclave/bits/safecrt.h>
 #include <openenclave/bits/safemath.h>
+#include <openenclave/corelibc/stdio.h>
 #include <openenclave/enclave.h>
 #include <openenclave/internal/calls.h>
 #include <openenclave/internal/raise.h>
 #include <stdlib.h>
 #include <string.h>
 #include "../common/sgx/revocation.h"
-
-/**
- * Validate and copy buffer to enclave memory.
- */
-static oe_result_t _copy_buffer_to_enclave(
-    uint8_t** dst,
-    size_t* dst_size,
-    const uint8_t* src,
-    size_t src_size)
-{
-    oe_result_t result = OE_FAILURE;
-    if (!src || src_size == 0 || !oe_is_outside_enclave(src, src_size) ||
-        dst == NULL || dst_size == NULL)
-        OE_RAISE(OE_INVALID_PARAMETER);
-
-    *dst = malloc(src_size);
-    if (*dst == NULL)
-        OE_RAISE(OE_OUT_OF_MEMORY);
-
-    OE_CHECK(oe_memcpy_s(*dst, src_size, src, src_size));
-    *dst_size = src_size;
-    result = OE_OK;
-
-done:
-    return result;
-}
+#include "internal_t.h"
 
 /**
  * Call into host to fetch revocation information.
  */
 oe_result_t oe_get_revocation_info(oe_get_revocation_info_args_t* args)
 {
     oe_result_t result = OE_FAILURE;
-    size_t host_args_buffer_size = sizeof(*args);
-    uint8_t* host_args_buffer = NULL;
-    oe_get_revocation_info_args_t* host_args = NULL;
-    oe_get_revocation_info_args_t tmp_args = {0};
-    uint8_t* p = NULL;
-    size_t crl_url_sizes[2] = {0};
-
-    if (args == NULL || args->num_crl_urls != 2 || args->crl_urls[0] == NULL ||
-        args->crl_urls[1] == NULL)
-        OE_RAISE(OE_FAILURE);
+    oe_get_revocation_info_args_t in;
+    oe_get_revocation_info_args_t out;
```

Comment:
Consider initializing the structs at definition.

Reply:
Again, I want to see some documentaiton about this.

Reply:
I researched this and I'm satisfied with "{ 0 }". Using that style.